Signup: Add a Store option to the design type step.

kwight opened the pull request:
![screen shot 2016-06-27 at 12 44 12 pm](https://cloud.githubusercontent.com/assets/349751/16393267/ef0bb0f8-3c64-11e6-993a-f63761c8bf7f.png)

Test live: https://calypso.live/?branch=add/signup-store-design

kwight commented:
cc/ @melchoyce @sirbrillig

melchoyce commented:
Getting an error when I try to load the calypso live branch:

```Error: ENOENT: no such file or directory, open '/home/ubuntu/data/gh-wp-calypso/logs/add/signup-store-design.log'```

sirbrillig commented:
At 1440x900, the bottom two options are mostly below the fold. We probably want to shrink them all so that the options don't go below the fold for most reasonable screen heights. 

<img width="819" alt="screen shot 2016-06-27 at 4 15 20 pm" src="https://cloud.githubusercontent.com/assets/2036909/16394307/8a35d92e-3c82-11e6-918f-e0cf03d98b86.png">

kwight commented:
@melchoyce I think you should be good now, maybe it needs some time to fire up ¯\_(ツ)_/¯ (I got it to eventually load).

melchoyce commented:
Cool, working for me now too 👍 

I agree it feels very big. Can we retain the same width we had previously? (320px, I believe)

melchoyce commented:
Also noticing there's a 1px gap between the store svg and the container. Do you know if that's on my side? I can edit the svg and resend, if so.

melchoyce commented:
Alternately, I tried out this, but it felt too small (and I was worried about translations):

![triforce e-commerce step 1 alt 2](https://cloud.githubusercontent.com/assets/2846578/16395081/3c1462e8-3c86-11e6-8690-d1a6afbc5f42.png)

kwight commented:
Keeping the cards the same size does look better; this is at 1440x900 now:

![screen shot 2016-06-27 at 2 03 56 pm](https://cloud.githubusercontent.com/assets/349751/16395705/23d78572-3c70-11e6-9fcb-ee179c157fdb.png)

kwight commented:
> Also noticing there's a 1px gap between the store svg and the container.

@melchoyce I haven't changed anything within the SVG; can you take a look?

kwight commented:
Re: i18n, @hewsut suggested Russian and German as good first checks – here they are:

![screen shot 2016-06-27 at 2 20 00 pm](https://cloud.githubusercontent.com/assets/349751/16396188/69a6cd40-3c72-11e6-9f0a-aa836e28aae8.png)

![screen shot 2016-06-27 at 2 19 26 pm](https://cloud.githubusercontent.com/assets/349751/16396191/6eaeb10e-3c72-11e6-8f82-079c5a020d21.png)

Ewww, line height weirdness. Already reported in #6110 .

kwight commented:
And Hebrew for good measure:

![screen shot 2016-06-27 at 3 26 32 pm](https://cloud.githubusercontent.com/assets/349751/16397833/9f3d12a8-3c7b-11e6-8fb9-b95bb59f7f08.png)

melchoyce commented:
Size looks good. Here's a fixed SVG: https://cloudup.com/imT8kmNm0Hl

kwight commented:
@melchoyce Done! @sirbrillig could you do a quick code sanity check, and then we should be good to turn this over.

kwight commented:
Hm, that new SVG doesn't look right, checking.

sirbrillig commented:
Code looks good. Nice work! 👍

sirbrillig commented:
Attempting to fix line-height in #6371

kwight commented:
And the final SVG update:

![screen shot 2016-06-28 at 10 32 14 am](https://cloud.githubusercontent.com/assets/349751/16426319/548d6a94-3d1d-11e6-993b-15472ddd6898.png)

melchoyce commented:
LGTM! 👍

kwight commented:
@meremagee OK, this is ready to go; the commit can just be `cherry-pick`ed by whomever is integrating it all into signup with the other step.

kwight commented:
@sirbrillig Good point; after Slack discussion, I've removed the shuffling, and we're keeping the order what it was in the [last screenshot](https://github.com/Automattic/wp-calypso/pull/6340#issuecomment-229126268) above.

melchoyce commented:
New store icon: https://cloudup.com/icMJSB6qbzy

kwight commented:
Updated (also changed the SVG's header background to match #E8F0F5 in the other SVGs:

![screen shot 2016-06-28 at 1 08 33 pm](https://cloud.githubusercontent.com/assets/349751/16430680/ab8f25c6-3d31-11e6-9e34-23cdeac52f42.png)

bisko commented on client/signup/steps/design-type-with-store/index.jsx:
```diff
@@ -0,0 +1,120 @@
+		return (
+			<div className="design-type-with-store">
+				<div className={ pressableWrapperClassName } >
+					<PressableStoreStep { ... this.props } ref={ this.onPressableStoreStepRef }/>
```
Wouldn't it be better to have this rendered only when it needs to be shown and not have it in the DOM at all times?

There is already similar logic in the Domains step in Signup and the pattern could possibly be reused here. Code can be found [here](https://github.com/Automattic/wp-calypso/blob/045b256fa4531b0ddc0f376b738e8fdac70a6d92/client/signup/steps/domains/index.jsx#L225).

michaeldcain replied:
Also, the `backUrl` bit in that same function would solve our broken back button when the Pressable sub-step is rendered.

coreh replied:
I'm rendering both simultaneously to enable a smooth CSS transition, like @shaunandrews's PR on the verticals survey: #6487.

coreh replied:
And for the back button, I planned on recreating the button provided by `StepWrapper` instead of relying on a URL, otherwise we'll need to add routes to the step. It's also what Survey is doing (though with more custom-looking UI)

michaeldcain commented:
LGTM. @coreh go ahead and squash and merge. If you want to open up a PR for the back button, I'll use #6438 to implement the A/B.
